diff -ur netpbm-10.73.43.orig/buildtools/libopt.c netpbm-10.73.43/buildtools/libopt.c
--- netpbm-10.73.43.orig/buildtools/libopt.c	2016-03-27 03:38:28.000000000 +0200
+++ netpbm-10.73.43/buildtools/libopt.c	2025-08-15 17:53:13.231710308 +0200
@@ -88,7 +88,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 
-typedef unsigned char bool;
+typedef unsigned char boolvar;
 #ifndef TRUE
 #define TRUE (1)
 #endif
