diff -Nur netpbm-9.12.orig/pnm/jbig/Makefile netpbm-9.12/pnm/jbig/Makefile
--- netpbm-9.12.orig/pnm/jbig/Makefile	Wed Feb 21 06:14:38 2001
+++ netpbm-9.12/pnm/jbig/Makefile	Thu Apr  5 21:39:27 2001
@@ -23,14 +23,14 @@
            $(PPMDIR)/$(LIBPPM) $(PGMDIR)/$(LIBPGM) $(PBMDIR)/$(LIBPBM) 
 
 BINARIES = jbigtopnm pnmtojbig
-OBJECTS = $(patsubst %, %.o, $(BINARIES)) jbig.o jbig_tab.o
+OBJECTS = $(patsubst %, %.o, $(BINARIES))
 
 MANUALS1 = $(BINARIES)
 
 all: $(BINARIES)
 
-$(BINARIES): %: %.o jbig.o jbig_tab.o $(NETPBMLIBS) $(LIBOPT)
-	$(LD) $(LDFLAGS) -o $@ $< jbig.o jbig_tab.o `$(LIBOPT) $(NETPBMLIBS)` \
+$(BINARIES): %: %.o $(NETPBMLIBS) $(LIBOPT)
+	$(LD) $(LDFLAGS) -o $@ $< -ljbig `$(LIBOPT) $(NETPBMLIBS)` \
 	  $(CDEBUG)
 
 
