--- netpbm-10.35.65/buildtools/configure.pl~	2009-06-26 03:35:42.000000000 +0200
+++ netpbm-10.35.65/buildtools/configure.pl	2009-07-15 08:47:35.271968502 +0200
@@ -1889,11 +1889,6 @@
         push(@Makefile_config, "CFLAGS += -fPIC\n");
         push(@Makefile_config, "LDSHLIB = -shared -fPIC\n");
         push(@Makefile_config, 'LDFLAGS += -Wl,+b,/usr/pubsw/lib', "\n");
-    } else {
-        # We don't know what to do here.  We used to (before 10.20) just
-        # just assume the compiler was gcc.  We know that the gcc stuff
-        # above does NOT work for HP native compiler.
-        push(@config_mk, "LDSHLIB =\n");
     }
 } elsif ($platform eq "AIX") {
     push(@Makefile_config, 'LDFLAGS = -L /usr/pubsw/lib', "\n");
