diff -urN netpbm-9.25/pbm/libpm.c netpbm-9.25.patch/pbm/libpm.c
--- netpbm-9.25/pbm/libpm.c	2002-02-24 23:37:29.000000000 +0100
+++ netpbm-9.25.patch/pbm/libpm.c	2004-08-06 01:47:36.000000000 +0200
@@ -644,6 +644,8 @@
     struct stat statbuf;
     FILE * original_file;
     FILE * seekable_file;
+#define TEMPFILE_NAME   "/tmp/netpbm.XXXXXX"
+    char fname[sizeof(TEMPFILE_NAME)];
 
     original_file = pm_openr((char *) name);
 
@@ -667,8 +669,10 @@
 
     if (seekable) {
         seekable_file = original_file;
-    } else {
-        seekable_file = tmpfile();
+    } else {	    
+/*        seekable_file = tmpfile();*/
+	    strcpy(fname, TEMPFILE_NAME);
+	    seekable_file = fdopen(mkstemp(fname), "w+");
 
         /* Copy the input into the temporary seekable file */
         while (!feof(original_file) && !ferror(original_file) 
diff -urN netpbm-9.25/ppm/ppmtompeg/parallel.c netpbm-9.25.patch/ppm/ppmtompeg/parallel.c
--- netpbm-9.25/ppm/ppmtompeg/parallel.c	2001-08-31 22:48:30.000000000 +0200
+++ netpbm-9.25.patch/ppm/ppmtompeg/parallel.c	2004-08-06 02:36:57.000000000 +0200
@@ -575,8 +575,10 @@
 
   if ( frameNumber != -1 ) {
     if ( separateConversion ) {
-      sprintf(fileName, "/tmp/foobar%d", machineNumber);
-      filePtr = fopen(fileName, "wb");
+/*      sprintf(fileName, "/tmp/foobar%d", machineNumber);*/
+#define TMPFILE_NAME "/tmp/ppmtompeg.XXXXXX"
+	    strcpy(fileName, TMPFILE_NAME);
+      filePtr = fdopen(mkstemp(fileName), "wb");
 
       /* read in stuff, SafeWrite to file, perform local conversion */
       do {
diff -urN netpbm-9.25/ppm/vidtoppm.c netpbm-9.25.patch/ppm/vidtoppm.c
--- netpbm-9.25/ppm/vidtoppm.c	2000-05-16 22:22:34.000000000 +0200
+++ netpbm-9.25.patch/ppm/vidtoppm.c	2004-08-06 02:18:55.000000000 +0200
@@ -149,6 +149,8 @@
   char ofname[256];
   int height, width;
   char command[256];
+#define TMPFILE_NAME "/tmp/vidtoppm.XXXXXX"
+  char tmpfilename[sizeof(TMPFILE_NAME)];
 
   ppm_init(&argc, argv);
 
@@ -245,7 +247,8 @@
 
 
     sprintf(ofname, "%s%d.ppm", obase, i);
-    outFile = fopen("/tmp/foobar", "wb");
+    strcpy(tmpfilename, TMPFILE_NAME);    
+    outFile = fdopen(mkstemp(tmpfilename), "wb");
     if (!outFile) {
       perror("Couldn't open output file.");
     }
@@ -263,8 +266,8 @@
 
     free(tdata);
 
-    sprintf(command, "rawtoppm %d %d < /tmp/foobar > %s",
-	    width, height, ofname);
+    sprintf(command, "rawtoppm %d %d < %s > %s",
+	    width, height, tmpfilename, ofname);
     system(command);
   }
 }
