--- netpbm-9.5/pnm/pstopnm.csh.pstopnm	Mon Jul 17 16:45:40 2000
+++ netpbm-9.5/pnm/pstopnm.csh	Mon Jul 17 17:04:08 2000
@@ -235,7 +235,7 @@
 	    # xmax and ymax
 	    set xres = `(echo "scale=4";echo "$xmax *72/$sx")|bc -l`
 	    set yres = `(echo "scale=4";echo "$ymax *72/$sy")|bc -l`
-	    set xres = `(echo "scale=4";echo "if($xres>$yres)$yres";echo "if($yres>$xres)$xres")|bc -l`
+	    set xres = `(echo "scale=4";echo "if($xres>$yres)$yres";echo "if($yres>$xres)$xres";echo "if($xres==$yres)$xres")|bc -l`
 	    set yres = $xres
 	    if ($?nocrop) then
 		# keep output file dimensions equal to xmax and ymax
