diff -urN netpbm-9.9/pgm/asciitopgm.c netpbm-9.9.new/pgm/asciitopgm.c
--- netpbm-9.9/pgm/asciitopgm.c	Tue Dec 19 18:00:05 2000
+++ netpbm-9.9.new/pgm/asciitopgm.c	Tue Dec 19 18:03:18 2000
@@ -118,7 +118,9 @@
 	    line_done:
 		    for (i = 0; i < cols; ++i)
 			grays[row][i] = maxval - (obuf[i] / divisor);
-		    memcpy (obuf, 0, cols * sizeof (int));
+                    /* 12/19/00 Phil Knirsch <pknirsch@redhat.com>:
+                       Fix: Use memset instead of totaly wrong memcpy. */
+		    memset (obuf, 0, cols * sizeof (int));
 		    i = 0;
 		    ++row;
 		    if ( row >= rows )
