--- netpbm-10.73.34/converter/ppm/ximtoppm.c.orig	2020-06-27 19:03:16.000000000 +0200
+++ netpbm-10.73.34/converter/ppm/ximtoppm.c	2021-01-19 20:53:46.388484792 +0100
@@ -116,9 +116,9 @@
 */
     header->bits_channel = atoi(a_head.bits_per_channel);
     header->alpha_flag = atoi(a_head.alpha_channel);
-    pm_asprintf(&header->author,  a_head.author);
-    pm_asprintf(&header->date,    a_head.date);
-    pm_asprintf(&header->program, a_head.program);
+    pm_asprintf(&header->author,  "%s", a_head.author);
+    pm_asprintf(&header->date,    "%s", a_head.date);
+    pm_asprintf(&header->program, "%s", a_head.program);
     /* Do double checking for bakwards compatibility */
     if (header->npics == 0)
         header->npics = 1;
