--- netpbm-10.23/lib/util/wordaccess.h.orig	2004-06-29 21:20:25.000000000 +0000
+++ netpbm-10.23/lib/util/wordaccess.h	2004-07-19 16:45:46.000000000 +0000
@@ -65,7 +65,7 @@
    exact same format as wordintBytes.  It's a simple matter of casting
    to fool the compiler.  
 */
-static __inline__ wordn
+static __inline__ wordint
 bytesToWordint(wordintBytes bytes) {
     return *((wordint *)bytes);
 }
